scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -589,6 +589,17 @@ def load_from_file(self, file, filter_platform=[]):
                     instance.status = status
                     instance.reason = reason
 
+                if self.quarantine:
+                    matched_quarantine = self.quarantine.get_matched_quarantine(
+                        instance.testsuite.id, platform.name, platform.arch, platform.simulation
+                    )
+                    if matched_quarantine and not self.options.quarantine_verify:
+                        instance.run = False
```

Comment:
don't need to set `instance.run` because you add filter (so this test will be skipped)

Reply:
you can extract that method to
```
def handle_quarantined_tests(self, instance: TestInstance, plat: Platform):
    if self.quarantine:
        matched_quarantine = self.quarantine.get_matched_quarantine(
            instance.testsuite.id, plat.name, plat.arch, plat.simulation
        )
        if matched_quarantine and not self.options.quarantine_verify:
            instance.add_filter("Quarantine: " + matched_quarantine, Filters.QUARANTINE)
        if not matched_quarantine and self.options.quarantine_verify:
            instance.add_filter("Not under quarantine", Filters.QUARANTINE)
```
and use: 
`self.handle_quarantined_tests(instance, platform)`
(remember also to use it in `apply_filters` method).